src/Microsoft.AspNet.Hosting/Internal/HostingEngine.cs

```diff
@@ -79,6 +80,8 @@ public virtual IDisposable Start()
                         contextAccessor.HttpContext = httpContext;
                         await application(httpContext);
                     }
+
+                    return httpContext;
```

Comment:
Is open question on whether the Context is the place to dispose of the features collection: https://github.com/aspnet/HttpAbstractions/blob/dev/src/Microsoft.AspNet.Http/DefaultHttpContext.cs#L172

Currently it is in Dispose; but the dispose never gets called as its not returned, so the features collection is never disposed?

Have raised related issue https://github.com/aspnet/Hosting/issues/359 as these have all been PRs where its slightly tangential.